### DISCUSSION TITLE
Set react variant based on rehype element class?

### DISCUSSION THREAD

**Discussion by swamidass** [system-ui/theme-ui#2163] [Q&A]
I'm using html source files for articles that  are converted with rehype-parse, some plugins, then rehype-react. Is there a way to set the react components' variant based on the classes assigned to the hast elements?

**Comment by lachlanjc** [comment]
Hey! For more technical help I'd need to understand more of the specific implementation details, but a few things come to mind:

1. If it's based on the HTML tags, the BaseStyles component will apply your `styles` object: https://theme-ui.com/api/#basestyles
2. If you want to generate a global class based on a variant, you could create a class with the Global component (https://theme-ui.com/guides/global-styles) with a value of `theme.styles.h1` or wherever your variant lives
3. You could use React's cloneElement to render each node as a Theme UI component

Does that help?

**Reply by swamidass** [reply]
Thanks for the note. I was wondering more about a situation where a div has a class, e.g.

<div class="my-special-class"> content </div>

Is there a way to style that div by mapping the class to a variant?

**Reply by hasparus** [reply]
> I was wondering more about a situation where a div has a class

You don't need the `variant` _feature_ here IMHO. You already have a variant _concept_, but it just comes in play based on the class.

Something like this should work:

```js
const variantBlue = { color: "blue" }
const variantRed = { color: "red" }

const style = {
  "&.blue": variantBlue,
  "&.red": variantRed
}
```

**Reply by swamidass** [reply]
Yes, in this case that could work. Thanks.
